src/components/controls/tables/FancyTable/FancyTable.js

```diff
@@ -0,0 +1,193 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import styles from './FancyTable.module.css';
+import { usePagination, useTable, useFilters } from 'react-table';
+import { Table } from 'react-bootstrap';
+import { flow, map, min, max } from 'lodash/fp';
+import PaginationControls from '../PaginationControls';
+import NoColumnFilter from '../column-filters/NoColumnFilter';
+
+function FancyTable({
+  // Data to be displayed in table
+  data,
+  // Column definitions for table
+  columns,
+  // Hidden columns
+  hiddenColumns = [],
+  // Factory for defaultColumn argument to useTable (see note)
+  makeDefaultColumn = () => {},
+  // Factory for filterTypes argument to useTable (see note)
+  makeFilterTypes = () => {},
+}) {
+  // Note: Several arguments to useTable are passed in from outside this
+  // component, which endeavours to be a bit general. Some of those arguments
+  // require memoizing with React.useMemo (see React Table documentation).
+  // Memoization can only be done at the top level of a React component;
+  // therefore we receive value factories, not memoized values, for these
+  // arguments, and memoize them here.
+  const {
+    // Basic table functionality
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    prepareRow,
+    visibleColumns,
+
+    // Pagination
+    // Instead of using `rows`, we use `page`,
+    // which has only the rows for the active page
+    page,
+    canPreviousPage,
+    canNextPage,
+    pageCount,
+    gotoPage,
+    nextPage,
+    previousPage,
+    setPageSize,
+
+    // Know what's in state
+    state: {
+      // Pagination
+      pageIndex,
+      pageSize
+    },
+  } = useTable(
+    {
+      columns,
+      data,
+      defaultColumn: React.useMemo(makeDefaultColumn, []),
+      filterTypes: React.useMemo(makeFilterTypes, []),
```

Comment:
Improper memoization: The factory should be constructed here from a non-factory function passed in. Thus:

```
filterTypes: React.useMemo(() => filterTypes(), [])
```